src/Security/OtpGenerator.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+/**
+ * Copyright 2014 SURFnet bv
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Surfnet\StepupBundle\Security;
+
+use Surfnet\StepupBundle\Exception\LogicException;
+use Surfnet\StepupBundle\Security\Exception\OtpGenerationRuntimeException;
+
+/**
+ * A stand-alone class for securely generating OTPs.
+ */
+final class OtpGenerator
+{
+    /**
+     * The characters used in the OTP. Must be a power of two characters long to ensure all characters have equal chance
+     * of being included.
+     */
+    const ALPHABET = 'ABCDEFGHJKLMNPQRSTUVWXYZ23456789';
+
+    /**
+     * Securely generate a 8-character OTP containing only characters from the OtpGenerator::ALPHABET constant.
+     * Based on https://gist.github.com/pmeulen/3dff8bab3227ed340dd1
+     *
+     * @return string
+     * @throws OtpGenerationRuntimeException
+     */
+    public static function generate()
+    {
+        $passwordLength = 8; // The length of the password to generate
+        $bitsPerValue = 5;  // The 32 possible characters in the Base32 alphabet can be represented in exactly 5 bits
+        $randomBytesRequired = (int) (($passwordLength * $bitsPerValue) / 8) + 1;
+        $cryptoStrong = false;
+        $randomBytes = openssl_random_pseudo_bytes($randomBytesRequired, $cryptoStrong); // Generate random bytes
+
+        if ($cryptoStrong === false) {
```

Comment:
FYI: The likelyhood of this ever occurring is 0 - this would be caught well before on dev or acc envs. This does not happen 1 in X amount of OTPs, either it can use strong crypto or it can't - depends on the likes of OS and available packages etc.